src/core/build_simulations.jl

```diff
@@ -77,21 +75,32 @@ function _build_stages(sim_ref::SimulationRef,
 
     system_to_file = get(kwargs, :system_to_file, true)
     mod_stages = Vector{_Stage}(undef, length(stages))
-
-    for (k, v) in stages
+    for (k, stage) in stages
         verbose && @info("Building Stage $(k)")
-        op_mod = OperationModel(DefaultOpModel, v.model, v.sys;
-                                optimizer = v.optimizer,
-                                parameters = true,
-                                verbose = verbose,
-                                kwargs...)
+        canonical = _build_canonical(stage.model.transmission,
+                                    stage.model.devices,
+                                    stage.model.branches,
+                                    stage.model.services,
+                                    stage.sys,
+                                    stage.optimizer,
+                                    verbose;
+                                    parameters = true,
+                                    kwargs...)
         stage_path = joinpath(sim_ref.models,"stage_$(k)_model")
         mkpath(stage_path)
-        write_op_model(op_mod, joinpath(stage_path, "optimization_model.json"))
-        system_to_file && IS.to_json(v.sys, joinpath(stage_path ,"sys_data.json"))
-        _populate_cache!(v.cache, op_mod)
-        mod_stages[k] = _Stage(k, op_mod, v.execution_count, v.chronology_ref, v.cache)
-        sim_ref.date_ref[k] = PSY.get_forecast_initial_times(v.sys)[1]
+        _write_canonical_model(canonical, joinpath(stage_path, "optimization_model.json"))
+        system_to_file && IS.to_json(stage.sys, joinpath(stage_path ,"sys_data.json"))
+        mod_stages[k] = _Stage(k,
```

Comment:
"k" is not very clear.  Is this stage_id?  stage_index?

Reply:
Changed to key, _Stages to have keys as a field.